proposals/extend_cli_for_generate_policies.md

```diff
@@ -0,0 +1,205 @@
+# Kyverno Design Proposal - Extend CLI test command for generate policies
+
+Author: [Shubham Nazare](https://github.com/shubham4443)
+
+PR: [#3456](https://github.com/kyverno/kyverno/pull/3456)
+
+## Contents
+
+- [Kyverno Design Proposal - Extend CLI test command for generate policies](#kyverno-design-proposal---extend-cli-test-command-for-generate-policies)
+  * [Introduction](#introduction)
+  * [Proposal](#proposal)
+    + [API Design Changes](#api-design-changes)
+    + [Method to test data object](#method-to-test-data-object)
+      - [Example: Add Network Policy](#example--add-network-policy)
+    + [Method to test clone object](#method-to-test-clone-object)
+      - [Example: Sync Secrets](#example--sync-secrets)
+    + [Namespaced Policies](#namespaced-policies)
+      - [Example: Policy Disruption Budget](#example--policy-disruption-budget)
+  * [Implementation details](#implementation-details)
+    + [Mock Generate-Request Controller](#mock-generate-request-controller)
+    + [Building the result](#building-the-result)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
+
+## Introduction
+
+Prior to this proposal, only validate and mutate policies were covered by the kyverno test CLI command. Generate policies were skipped by the CLI. This document proposes an approach for the test command to cover generate policies as well.
+
+## Proposal
+
+When using a generate rule, the origin resource can be either an existing resource defined within Kubernetes, or a new resource defined in the rule itself. When the origin resource is a pre-existing resource the clone object is used. When the origin resource is a new resource defined within the manifest of the rule, the data object is used. Both rules are mutually exclusive, and only one will be specified in a rule. Both objects require a different approach for testing through CLI command.
```

Comment:
We usually have the trigger and the target resources in a generate policy. What is this origin resource?

Reply:
Its the same as target resource. In the documentation https://kyverno.io/docs/writing-policies/generate/ it is mentioned as origin resource so I used the same word.